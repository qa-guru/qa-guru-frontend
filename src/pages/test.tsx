import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
