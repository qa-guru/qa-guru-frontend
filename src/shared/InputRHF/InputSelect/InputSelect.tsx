import React from "react";
import { Controller, FieldValues } from "react-hook-form";
import { InputLabel, MenuItem, Select } from "@mui/material";
import { IFormInputProps } from "../Input.types";

const InputSelect = <T extends FieldValues>({
  control,
  placeholder,
  name,
  options,
  defaultValue,
  onChange,
  disabled = false,
}: IFormInputProps<T>) => {
  return (
    <>
      <InputLabel>{placeholder}</InputLabel>
      <Controller
        name={name}
        control={control}
        defaultValue={defaultValue}
        render={({ field: { value, onChange: handleChange } }) => (
          <Select
            label={placeholder}
            onChange={(event) => {
              handleChange(event);
              if (onChange) {
                onChange(event.target.value as string);
              }
            }}
            value={value}
            disabled={disabled}
          >
            {options?.map((option, index) => (
              <MenuItem key={index} value={option.value!}>
                {option.label}
              </MenuItem>
            ))}
          </Select>
        )}
      />
    </>
  );
};

export default InputSelect;
