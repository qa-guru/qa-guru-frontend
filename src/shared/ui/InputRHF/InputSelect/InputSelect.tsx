import React from "react";
import { Controller } from "react-hook-form";
import { IFormInputProps } from "../Input.types";
import { FormControl, InputLabel, MenuItem, Select } from "@mui/material";

const InputSelect: React.FC<IFormInputProps> = ({
  control,
  placeholder,
  name,
  content,
  defaultValue,
}) => {
  const generateMenuItem = () => {
    return content.map((option: string) => (
      <MenuItem key={option} value={option}>
        {option}
      </MenuItem>
    ));
  };

  return (
    <FormControl>
      <InputLabel>{placeholder}</InputLabel>
      <Controller
        name={name}
        control={control}
        defaultValue={defaultValue}
        render={({ field: { value, onChange } }) => (
          <Select label={placeholder} onChange={onChange} value={value}>
            {generateMenuItem()}
          </Select>
        )}
      />
    </FormControl>
  );
};

export default InputSelect;
