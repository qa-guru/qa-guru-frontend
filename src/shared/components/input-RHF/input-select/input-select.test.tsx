import { MemoryRouter } from "react-router-dom";
import { FormProvider, useForm } from "react-hook-form";
import { render } from "test/utils-test";
import InputSelect from "./input-select";

describe("InputSelect", () => {
  it("renders correctly", () => {
    function TestComponent() {
      const methods = useForm();
      return (
        <MemoryRouter>
          <FormProvider {...methods}>
            <InputSelect
              control={methods.control}
              name="testDate"
              placeholder="testDate"
              options={[{ value: "select", label: "label" }]}
              defaultValue="testDate"
              onChange={() => {}}
              disabled={false}
            />
          </FormProvider>
        </MemoryRouter>
      );
    }

    const { asFragment } = render(<TestComponent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
