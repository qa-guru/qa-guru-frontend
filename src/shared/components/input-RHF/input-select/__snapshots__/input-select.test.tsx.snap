// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputSelect > renders correctly 1`] = `
<DocumentFragment>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1r395si-MuiFormLabel-root-MuiInputLabel-root"
  >
    testDate
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-d4zdg1-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-1c0lsac-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      role="button"
      tabindex="0"
    >
      <span
        class="notranslate"
      >
        â€‹
      </span>
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      tabindex="-1"
      value="testDate"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-yjsfm1"
      >
        <span>
          testDate
        </span>
      </legend>
    </fieldset>
  </div>
</DocumentFragment>
`;
